kornia/contrib/sam/architecture/sam.py

```diff
@@ -0,0 +1,298 @@
+                encoder_depth=24,
+                encoder_num_heads=16,
+                encoder_global_attn_indexes=(5, 11, 17, 23),
+            )
+
+        elif model_type == SamModelType.vit_h:
+            model = _build_sam(
+                encoder_embed_dim=1280,
+                encoder_depth=32,
+                encoder_num_heads=16,
+                encoder_global_attn_indexes=(7, 15, 23, 31),
+            )
+        else:
+            raise NotImplementedError('Unexpected model type. Should be vit_b, vit_l or vit_h.')
+
+        return model
+
+    @staticmethod
+    def from_pretrained(
+        model_type: str | int | SamModelType, checkpoint: str | None = None, device: torch.device | None = None
+    ) -> Sam:
+        """This builds the desired SAM model, load the checkpoint and move the model into the desired device.
+
+        Args:
+            model_type: the available models are:
+
+                - 0, 'vit_h' or :func:`kornia.contrib.sam.SamModelType.vit_h`
+                - 1, 'vit_l' or :func:`kornia.contrib.sam.SamModelType.vit_l`
+                - 2, 'vit_b' or :func:`kornia.contrib.sam.SamModelType.vit_b`
+
+            checkpoint: The url or filepath for the respective checkpoint
+            device: The desired device to load the weights and move the model
+
+        Returns:
+            The respective SAM model
+
+        Example:
+            >>> # Input should be a RGB batched image
+            >>> inpt = torch.randint(0, 255, (1, 3, 384, 384)).float()
+            >>> inpt_after_resize = kornia.geometry.resize(inpt, (256, 256))
+            >>> sam_model = Sam.from_pretrained('vit_b', 'cpu', checkpoint=None)
+            >>> # Embed prompts
+            >>> sparse_embeddings, dense_embeddings = sam_model.prompt_encoder(points=None, boxes=None, masks=None)
+            >>> # Preprocess input (expected max size to be 1024)
+            >>> input_image = sam_model.preprocess(inpt_after_resize)
+            >>> # Predict masks
+            >>> low_res_masks, iou_predictions = sam_model.mask_decoder(
+            ...    image_embeddings=sam_model.image_encoder(input_image),
+            ...    image_pe=sam_model.prompt_encoder.get_dense_pe(),
+            ...    sparse_prompt_embeddings=sparse_embeddings,
+            ...    dense_prompt_embeddings=dense_embeddings,
+            ...    multimask_output=True,
+            ... )
+            >>> # Upscale the masks to the original image resolution
+            >>> input_shape = (inpt_after_resize.shape[-2], inpt_after_resize.shape[-1])
+            >>> original_shape = (inpt.shape[-2], inpt.shape[-1])
+            >>> masks = sam_model.postprocess_masks(low_res_masks, input_shape, original_shape)
+            >>> # If wants to have a binary mask
+            >>> masks = masks > sam_model.mask_threshold
+        """
+
+        model = Sam.build(model_type)
+
+        if checkpoint:
+            model.load_checkpoint(checkpoint, device)
+
+        if device:
```

Comment:
I would remove this and leave to the user as standard pytorch practice to avoid api complexity

Reply:
sure, I had put it because I started to annotate with core.Device, but the module did not have the support integer device typing